administration/src/cards/createCards.ts

```diff
@@ -22,37 +16,34 @@ export class CreateCardsError extends Error {
   }
 }
 
-async function createCards(client: ApolloClient<object>, project: string, activationCodes: Codes, region: Region) {
-  const cards: CardGenerationModelInput[] = await Promise.all(
-    activationCodes.map(async code => {
-      const codeType = code instanceof DynamicActivationCode ? CodeType.Dynamic : CodeType.Static
-      const cardInfoHash = await hashCardInfo(code.info!, code.pepper)
-      const expirationDay = code.info!.expirationDay
-      const startDay = code.info!.extensions?.extensionStartDay?.startDay
-      const activationSecretBase64 =
-        code instanceof DynamicActivationCode ? uint8ArrayToBase64(code.activationSecret) : null
-      return {
-        cardExpirationDay: expirationDay ?? null, // JS number can represent integers up to 2^53, so it can represent all values of an uint32 (protobuf)
-        cardInfoHashBase64: uint8ArrayToBase64(cardInfoHash),
-        activationSecretBase64: activationSecretBase64,
-        regionId: region.id,
-        codeType,
-        cardStartDay: startDay,
-      }
-    })
-  )
-  const result = await client.mutate<AddCardsMutation, AddCardsMutationVariables>({
-    mutation: AddCardsDocument,
-    variables: { cards, project },
+type CreateCardsResult = { dynamicActivationCodes: DynamicActivationCode[], staticVerificationCodes: StaticVerificationCode[] }
+
+async function createCards(client: ApolloClient<object>, cardInfos: CardInfo[], staticCodes: boolean): Promise<CreateCardsResult> {
+  const cards = cardInfos.map(cardInfo => {
+    const encodedCardInfoBase64 = uint8ArrayToBase64(cardInfo.toBinary())
+    return { encodedCardInfoBase64, generateDynamicActivationCode: true, generateStaticVerificationCode: staticCodes }
+  })
+  const result = await client.mutate<CreateCardsMutation, CreateCardsMutationVariables>({
+    mutation: CreateCardsDocument,
+    variables: { cards },
   })
 
   if (result.errors) {
     const { title } = getMessageFromApolloError(new ApolloError({ graphQLErrors: result.errors }))
     throw new CreateCardsError(title)
   }
-  if (!result.data?.success) {
+  if (!result.data) {
     throw new CreateCardsError('Beim erstellen der Karte(n) ist ein Fehler aufgetreten.')
   }
+
+  return result.data.cards.reduce<CreateCardsResult>((acc, card) => {
+    const dynamicActivationCode = card.dynamicActivationCodeBase64 ? [DynamicActivationCode.fromBinary(base64ToUint8Array(card.dynamicActivationCodeBase64))] : []
+    const staticVerificationCode = card.staticVerificationCodeBase64 ? [StaticVerificationCode.fromBinary(base64ToUint8Array(card.staticVerificationCodeBase64))] : []
+    return {
+      dynamicActivationCodes: [...acc.dynamicActivationCodes, ...dynamicActivationCode],
+      staticVerificationCodes: [...acc.staticVerificationCodes, ...staticVerificationCode]
+    }
+  }, { dynamicActivationCodes: [], staticVerificationCodes: [] })
```

Comment:
We could define a much better type for this. But I think it makes sense to discuss this shortly in the team.